### ISSUE TITLE
Cannot use RiverMapper & pyDEM in the same script

### ISSUE THREAD

**Issue by SorooshMani-NOAA** [schism-dev/RiverMeshTools#1]
@feiye-vims while you're working on the latest generated mesh to see if it's useful or not, I started using RiverMapper and pyDEM to learn a bit more about them. To start I tried both serial and parallel sample for each of them successfully.

My next step was to try to combine the two serial scripts so that I end up with a single script that gets DEM as input and outputs river polygons. This is where I ran into an issue.

After some digging I realized that the issue stems from this line:
https://github.com/schism-dev/RiverMeshTools/blob/4ee6cd1cddb0df349dc20960b0e46710972eb578/RiverMapper/RiverMapper/make_river_map.py#L35

Basically setting this error handling in RiverMapper affects the results of pyDEM. So when at the top of myscript I import:
```python
from pyDEM.dem import dem
from RiverMapper.make_river_map import make_river_map
```

I run into an issue during the calculation of the DEM thalwegs:
```python
        dem_obj.compute_river(acc_limit=catchment_area)
        gdf = dem_obj.write_shapefile(npt_smooth=thalweg_smoothing)
        gdf.to_file(out_stem)
```

So I was wondering if there's any reason why this error handling is set for `numpy` errors?

**Comment by SorooshMani-NOAA** [comment]
I'm not sure if this just means we shouldn't set the error handling or we should fix the pyDEM's failure as a result of setting this error handling. If I comment that line in `make_river_map.py`, after calculating the thalwegs, when trying to create river map I get the following:
```
  File "/home/smani/workarea/sandbox/RiverMeshTools/RiverMapper/RiverMapper/make_river_map.py", line 1255, in make_river_map
    S, is_new_cache = Tif2XYZ(tif_fname=tif_fname, cache=i_DEM_cache)
  File "/home/smani/workarea/sandbox/RiverMeshTools/RiverMapper/RiverMapper/river_map_tif_preproc.py", line 62, in Tif2XYZ
    cache_name = tif_fname + '.pkl'
TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'
```

**Comment by feiye-vims** [comment]
Hi Soroosh, I'm glad you are testing the tools.

I set "np.seterr(all='raise') " to catch any error during numpy calculation such as division by zero or other invalid values. It is werid it can cause a different output from pyDEM without actually throwing an error in pyDEM. I can ask Linlin to check what caused the problem.

In the meantime, I'll just comment this line out, since make_river_map never throws this error in my previous tests.

We're still in the process of the annual update for STOFS-3D, so I didn't have a chance to get back to the grid problem, which I don't think is a big issue, it's just I never met a similar problem before. Hope you can allow a few more days.

**Comment by SorooshMani-NOAA** [comment]
Hi, no problem! OK, I'll also let you know if I figured out why pyDEM is doing this.

> In the meantime, I'll just comment this line out

Thank you, I'll let you know if during my tests I notice anything strange in RiverMapper as a results of commenting this out.

@cuill in my case the easiest way to see the issue is by adding
```diff
+ import numpy as np
+ np.seterr(all='raise') 
```
after importing all other packages in the `pyDEM_Samples/Serial/run_serial.py` example case.

**Comment by cuill** [comment]
@SorooshMani-NOAA Thanks for the information. I'll take a look.

**Comment by cuill** [comment]
@SorooshMani-NOAA You can pull the latest changes and test now. The issue with NaN values was also resolved. It should only have feature_id not reach_id.

**Comment by SorooshMani-NOAA** [comment]
@cuill I'm using the latest code from the repo, but I still get the same error when I set `np.seterr(all='raise') ` at the top of my script. The error happens during writing the dataframe to file due to invalid coordinates:
```
Traceback (most recent call last):
  File "/mnt/c/Users/Soroosh.Mani/WorkArea/Development/RiverMapper/own_test/dem_to_rivermap_parallel.py", line 315, in <module>
    main()
  File "/mnt/c/Users/Soroosh.Mani/WorkArea/Development/RiverMapper/own_test/dem_to_rivermap_parallel.py", line 270, in main
    shp_files = extract_thalwegs_mpi_wrap(
  File "/mnt/c/Users/Soroosh.Mani/WorkArea/Development/RiverMapper/own_test/dem_to_rivermap_parallel.py", line 75, in extract_thalwegs_mpi_wrap
    dist_gen_files = extract_thalwegs(
  File "/mnt/c/Users/Soroosh.Mani/WorkArea/Development/RiverMapper/own_test/dem_to_rivermap_parallel.py", line 130, in extract_thalwegs
    gdf.to_file(out_stem)
  File "/home/smani/miniconda3/envs/meshdev/lib/python3.10/site-packages/geopandas/geodataframe.py", line 1263, in to_file
    _to_file(self, filename, driver, schema, index, **kwargs)
  File "/home/smani/miniconda3/envs/meshdev/lib/python3.10/site-packages/geopandas/io/file.py", line 539, in _to_file
    _to_file_fiona(df, filename, driver, schema, crs, mode, **kwargs)
  File "/home/smani/miniconda3/envs/meshdev/lib/python3.10/site-packages/geopandas/io/file.py", line 568, in _to_file_fiona
    colxn.writerecords(df.iterfeatures())
  File "/home/smani/miniconda3/envs/meshdev/lib/python3.10/site-packages/fiona/collection.py", line 549, in writerecords
    self.session.writerecs(records, self)
  File "fiona/ogrext.pyx", line 1413, in fiona.ogrext.WritingSession.writerecs
RuntimeError: GDAL Error: Coordinates with non-finite values are not allowed. Failed to write record: <fiona.model.Feature object at 0x7fa3aca5c4c0>
```

**Comment by cuill** [comment]
@SorooshMani-NOAA Yes, I just got this error, too. In the previous test, I accidentally comment out np.seterr(all='raise') which did not raise an error. I'll look into this again when I have time.

**Comment by cuill** [comment]
https://github.com/OSGeo/gdal/issues/3542
